### ISSUE TITLE
Torch Perplexity Nightly CI Failing

### ISSUE THREAD

**Issue by Alex-Vasile** [nod-ai/shark-ai#1410]
Llama perplexity with torch has been failing by producing NaNs on nightly CI since April 30 (https://github.com/nod-ai/shark-ai/actions/runs/14774306957). It started failing sometime between 6513fe4 and 43e5545 and is currently failing. 

However, if I run it locally using the exact command in [ci-sharktank-nightly.yml](https://github.com/nod-ai/shark-ai/blob/d8887b53ed1f7fb8dc2cfea9eb4a98ed8ef8cefd/.github/workflows/ci-sharktank-nightly.yml#L199-L202), it passes.

It also passes, **at least the numerics do without creating NaNs**, on CI if the failed test is rerun on its own and not with the IREE perplexity tests run in parallel with it. The github action still fails since the subsequent "Deploy to GitHub Pages" step fails with the following error: `Error: Action failed with "not found deploy key or tokens"`.

**Comment by Alex-Vasile** [comment]
The same issue can be consistently reproduced by running two torch perplexity runs in parallel.

Doing that, this issue has been present since #1109, when we started running the perplexity tests on GPU.

Ran with `torch==2.5.1+rocm6.2`.

Though I don't know why the something in https://github.com/nod-ai/shark-ai/commit/6513fe4f5506989898e344a3091d000d6dcd303d through https://github.com/nod-ai/shark-ai/commit/43e554503b9c1a39207bfddff550c2bc5f10c133 has started to consistently hit this issue on the nightly CI.

**Comment by archana-ramalingam** [comment]
Adding context here:

Since switching eager mode perplexity tests to gpu on #1109, it has been producing nan logits and failing non-deterministically for both fp16 and fp8 llama models.

For example, on the below nightly run, switching between various attempts (on top right) shows attempts passing and failing for the same sharktank/IREE/torch versions, proving the non-deterministic nature of this failure.

Latest nightly run: https://github.com/nod-ai/shark-ai/actions/runs/14904900718/job/41904880860

Note: Reverting back to cpu is not a choice as fp8+CPU test takes upwards of 30 hrs to complete.

**Comment by archana-ramalingam** [comment]
Apart from Nan ppl scores in eager mode, in IREE mode, ppl scores shoots up sometimes. Will need to triage if both these issues are related. 
For example in a recent IREE ppl nightly run: 
1st attempt [failed](https://github.com/nod-ai/shark-ai/actions/runs/14927451130/job/41935443656#step:6:354) and 2nd [passed](https://github.com/nod-ai/shark-ai/actions/runs/14927451130/job/41964990336#step:6:73)in this workflow run. 

In torch eager the issue manifests as nan logits/ppl and in IREE the ppl shoots up. This behavior at least has been consistent across the past events and affects both fp16 and fp8 llama models.

**Comment by archana-ramalingam** [comment]
Observed nan results from Flux test running with IREE on GPU recently. 
Logs: https://github.com/nod-ai/shark-ai/actions/runs/14927451130/job/42242790419#step:6:76

**Comment by archana-ramalingam** [comment]
One reason for Nans: https://github.com/nod-ai/shark-ai/issues/1549
